packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -85,7 +87,7 @@ export async function fetchSnapshot(
 		url: string,
 		fetchOptions: { [index: string]: any },
 	) => Promise<IOdspResponse<unknown>>,
-): Promise<ISnapshotContents> {
+): Promise<ISnapshotContents | IPartialSnapshotWithContents> {
```

Comment:
Not sure if it matters a lot, but ISnapshotContents  looks very similar to IPartialSnapshotWithContents.
Given that ISnapshotContents is runtime only, driver only concept, we could massage it to look like IPartialSnapshotWithContents (if we want) and thus use only ISnapshotContents here. Not sure if it helps, but it's something to consider.

ISnapshotContents.latestSequenceNumber is missing on IPartialSnapshotWithContents (I know, I provided some comments before about it :)). That means that in places where we use ISnapshotContents | IPartialSnapshotWithContents, we likely can't rely on such concept. It looks like it's used only in convertToCompactSnapshot(), which is used only in test code.

It's provided by ODSP. I think we need to make a call - either it's useful concept and keep it on both interfaces, or drop from both.